front_end/panels/ai_chat/agent_framework/ConfigurableAgentTool.ts

```
@@ -412,27 +411,72 @@ export class ConfigurableAgentTool implements Tool<ConfigurableAgentArgs, Config
 
     // Initialize
     const maxIterations = this.config.maxIterations || 10;
-    const modelName = typeof this.config.modelName === 'function'
-      ? this.config.modelName()
-      : (this.config.modelName || AIChatPanel.instance().getSelectedModel());
+    
+    // Parse execution context first
+    const callCtx = (_ctx || {}) as {
+      provider?: import('../LLM/LLMTypes.js').LLMProvider;
```

ESLint

[error] `import()` type annotations are forbidden.

(@typescript-eslint/consistent-type-imports)